lib/monad.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+// id :: a -> a
+function id(x) {
+  return x;
+}
+
+// curry :: ((a, b) -> c) -> a -> b -> c
+function curry(f, ...args1) {
+  return (...args2) => f(...args1, ...args2);
+}
+
+function withCb(func) {
+  return (...argsCb) => {
+    const callback = argsCb[argsCb.length - 1];
+    const args = argsCb.slice(0, argsCb.length - 1);
+    return func(args, callback);
+  };
+}
+
+// fmap :: Functor f => (a -> b) -> f a -> f b
+function fmap(f, asyncF) {
+  const asyncF2 = withCb((args, callback) => of(f(...args))(callback));
+  return chain(asyncF, asyncF2);
+}
+
+// pure :: Applicative f => f (a -> b) -> f a -> f b
+function of(...args) {
```

Comment:
Cant understand purpose of this function. It's contract: `of(...args)(callback)` but why don't simply `callback(...args)` ?